ansible/roles/controller/tasks/deploy.yml

```diff
@@ -48,17 +56,17 @@
 
 - name: prepare controller ports
   set_fact:
-    ports_to_expose: ["{{ controller.basePort + groups['controllers'].index(inventory_hostname) }}:8080", "{{ controller.akka.cluster.basePort + groups['controllers'].index(inventory_hostname) }}:{{ controller.akka.cluster.bindPort }}"]
+    ports_to_expose: ["{{ controller.basePort + (controller_index | int) }}:8080", "{{ controller.akka.cluster.basePort + (controller_index | int) }}:{{ controller.akka.cluster.bindPort }}"]
```

Comment:
Is it possible to do the `int` conversation already on defining the variable? Or would this result in issues on gernerating the name-String?

Reply:
@cbickel, unfortunately defining the `controller_index` as an `int` results in the variable type being `unicode`. See below.

Define `controller_index` as an int:
```
- name: get controller index
  set_fact:
    controller_index: "{{ groups['controllers'].index(inventory_hostname) | int }}"
```

Attempting to add `controller_index` to an int:
```
{{ controller.basePort + controller_index }}
```

Results in this error:
```
fatal: [192.168.99.100]: FAILED! => {"failed": true, "msg": "Unexpected templating type error occurred on ({{ controller.basePort + controller_index }}:8080): unsupported operand type(s) for +: 'int' and 'unicode'"}
```

Apparently using `set_fact` always produces a string: https://github.com/ansible/ansible/issues/4170.